pkg/tls/certificatemanagement/ca.go

```diff
@@ -0,0 +1,596 @@
+}
+
+// CreateCertificateManager creates a signer of new certificates and has methods to retrieve existing KeyPairs and Certificates. If a user
+// brings their own secrets, CertificateManager will preserve and return them.
+func CreateCertificateManager(cli client.Client, installation *operatorv1.InstallationSpec, clusterDomain string) (CertificateManager, error) {
+	var cryptoCA *crypto.CA
+	var caSecret *corev1.Secret
+	var csrImage string
+	var certificateManagement *operatorv1.CertificateManagement
+	if installation != nil && installation.CertificateManagement != nil {
+		certificateManagement = installation.CertificateManagement
+		imageSet, err := imageset.GetImageSet(context.Background(), cli, installation.Variant)
+		if err != nil {
+			return nil, err
+		}
+		csrImage, err = components.GetReference(
+			components.ComponentCSRInitContainer,
+			installation.Registry,
+			installation.ImagePath,
+			installation.ImagePrefix,
+			imageSet,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		caSecret = &corev1.Secret{
+			TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      CASecretName,
+				Namespace: common.OperatorNamespace(),
+			},
+			Data: map[string][]byte{
+				corev1.TLSCertKey: certificateManagement.CACert,
+			},
+		}
+	} else {
+		caSecret = &corev1.Secret{}
+		err := cli.Get(context.Background(), types.NamespacedName{
+			Name:      CASecretName,
+			Namespace: common.OperatorNamespace(),
+		}, caSecret)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return nil, err
+		}
+		if caSecret == nil ||
+			len(caSecret.Data) == 0 ||
+			len(caSecret.Data[corev1.TLSPrivateKeyKey]) == 0 ||
+			len(caSecret.Data[corev1.TLSCertKey]) == 0 {
+			cryptoCA, err = tls.MakeCA(rmeta.TigeraOperatorCAIssuerPrefix)
+			if err != nil {
+				return nil, err
+			}
+			caSecret, err = getSecretFromTLSConfig(cryptoCA.Config, CASecretName, common.OperatorNamespace())
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			cryptoCA, err = crypto.GetCAFromBytes(caSecret.Data[corev1.TLSCertKey], caSecret.Data[corev1.TLSPrivateKeyKey])
+		}
+	}
+	x509Cert, err := parseCertificate(caSecret.Data[corev1.TLSCertKey])
+	if err != nil {
+		return nil, err
+	}
+	return &certificateManager{
+		CA: cryptoCA,
+		keyPair: keyPair{
+			csrImage:              csrImage,
+			Certificate:           x509Cert,
+			secret:                caSecret,
+			clusterDomain:         clusterDomain,
+			CertificateManagement: certificateManagement,
+		},
+	}, nil
+}
+
+func (cm *certificateManager) AddToStatusManager(statusManager status.StatusManager, namespace string) {
+	if cm.CertificateManagement() != nil {
+		statusManager.AddCertificateSigningRequests(namespace, map[string]string{"k8s-app": namespace})
+	} else {
+		statusManager.RemoveCertificateSigningRequests(namespace)
+	}
+}
+
+// GetOrCreateKeyPair returns a KeyPair. If one exists, some checks are performed. Otherwise, a new KeyPair is created.
```

Comment:
Could you please elaborate on the checks, I think it's important to outline that the given DNS names are used as validation of the certificate if it exists.